invoicer_project/static/js/add_edit_item.js

```diff
@@ -0,0 +1,166 @@
+import {nameField, priceField, amountInStockField, barcodeField, additionalFieldsContainer, amountAdditionalFieldsContainer} from './utils_items.js'
+import { removeAllErrorAttributes } from './validation_utils.js';
+
+document.querySelectorAll('.dropdown_list').forEach(function (dropdownWrapper) {
+    const dropdownBtn = dropdownWrapper.querySelector('.dropdown__button');
+    const dropdownArrow = dropdownWrapper.querySelector(".arrow-up");
+    const dropdownList = dropdownWrapper.querySelector('.dropdown__list');
+    const dropdownItems = dropdownList.querySelectorAll('.dropdown__list-item');
+    const dropdownInput = dropdownWrapper.querySelector('.dropdown__input_hidden');
+    dropdownBtn.addEventListener('click', function () {
+        dropdownArrow.classList.toggle("arrow-up");
+        dropdownArrow.classList.toggle("arrow-up-active");
+        dropdownList.classList.toggle('dropdown__list_visible');
+        this.classList.toggle('dropdown__button_active');
+        this.parentElement.classList.toggle('border_dropdown');
+    });
+
+    dropdownItems.forEach(function (listItem) {
+        listItem.addEventListener('click', function (e) {
+            dropdownItems.forEach(function (el) {
+                el.classList.remove('dropdown__list-item_active');
+            })
+            e.target.classList.add('dropdown__list-item_active');
+            dropdownBtn.innerText = this.innerText;
+            dropdownInput.setAttribute("value", this.dataset.value);
+            input_currency_table();
+            input_basic_unit_table();
+            dropdownArrow.classList.add("arrow-up");
+            dropdownArrow.classList.remove("arrow-up-active");
+            dropdownList.classList.remove('dropdown__list_visible');
+        })
+    });
+
+    document.addEventListener('click', function (e) {
+        if (e.target !== dropdownBtn) {
+            dropdownArrow.classList.add("arrow-up");
+            dropdownArrow.classList.remove("arrow-up-active");
+            dropdownBtn.classList.remove('dropdown__button_active');
+            dropdownList.classList.remove('dropdown__list_visible');
+        }
+    });
+})
+
+
+const inputCurrency = document.getElementById("currency");
+const inputBasicUnit = document.getElementById("basic_unit");
+const inputPriceVal = document.getElementById("price_val");
+const inputCurrencyVal = document.getElementById("currency_val");
+const inputBasicUnitVal = document.getElementById("basic_unit_val");
+const inputAmountInStockVal = document.getElementById("amount_in_stock_val");
+const inputNameVal = document.getElementById("name_val");
+const inputBarcodeVal = document.getElementById("barcode_val");
+const amountAdditionalUnitField = document.querySelectorAll(".amount_additional_unit_field");
+const additionalUnits = document.querySelectorAll(".additional_unit_field"); 
+const additionalUnitCell = document.querySelectorAll(".additional_unit_cell");
+
+let numOfRows = 0;
```

Comment:
Could we remove global variables?

Reply:
We use the variable numOfRows many times in the code and change it, so it's convenient to make it global, other variables of this type were removed in utils.